### PR TITLE
Void Game

### PR DESCRIPTION
**Pull request by kev306**

### CODE REVIEW

File: src/gameplay/voidGame.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Game from './game';
+
+export class VoidGame {
+  game: Game;
+
+  private playersVotedVoid: Set<string> = new Set();
```
Comment (kev306) [inline src/gameplay/voidGame.ts:6]: `playersVoted`

File: src/gameplay/voidGame.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  constructor(game: Game) {
+    this.game = game;
+  }
+
+  voteVoidGame(username: string, votesNeeded: number): number {
```
Comment (kev306) [inline src/gameplay/voidGame.ts:12]: Keep game or keep votesNeeded. If we keep votesNeeded, move to constructor

File: src/gameplay/game.ts [hunk 3]
```diff
@@ -1988,6 +1995,41 @@ class Game extends Room {
...
+    );
+
+    const s = numVoted > 1 ? 's have' : ' has';
+
+    this.sendText(
+      `${numVoted} player${s} voted to void the game. ${votesNeeded} votes needed.`,
+      'server-text',
+    );
+
+    if (numVoted >= votesNeeded) {
```
Comment (kev306) [inline src/gameplay/game.ts:2024]: `if (numVoted >= votesNeeded) ` lets move this logic into the voidGameVoteTracker

File: src/gameplay/voidGame.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    this.playersVotedVoid.add(username.toLowerCase());
+    return this.playersVotedVoid.size;
+  }
+}
```
Comment (kev306) [inline src/gameplay/voidGame.ts:16]: Add tests

File: src/sockets/sockets.ts [hunk 5]
```diff
@@ -951,6 +951,21 @@ export const userCommands = {
     },
   },
 
+  voteVoidGame: {
+    command: 'voteVoidGame',
```
Comment (kev306) [inline src/sockets/sockets.ts:955]:
```suggestion
    command: 'voidgame',
```

File: src/sockets/sockets.ts [hunk 6]
```diff
@@ -951,6 +951,21 @@ export const userCommands = {
...
+    help: '/voteVoidGame: Vote to void a game. Requires number_of_resistance + 1 votes.',
+    run(data, senderSocket) {
+      if (!senderSocket.request.user.inRoomId) {
+        senderSocket.emit('messageCommandReturnStr', {
+          message: 'You must be in a room to use /voteVoidGame.',
+          classStr: 'server-text',
+        });
```
Comment (kev306) [inline src/sockets/sockets.ts:962]: return

File: src/gameplay/voidGameTracker.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+export class VoidGameTracker {
+  private game: Game;
+  private playersVoted: Set<string> = new Set();
+
+  constructor(game: Game) {
+    this.game = game;
+  }
+
+  // Adds player to playersVoted set
+  // Returns true if numPlayersVoted > votesNeeded
```
Comment (vck3000) [inline src/gameplay/voidGameTracker.ts:13]: Can you remove line 12 and update line 13 to be something like:

> Returns true if the game should now be voided

Comment (kev306) [reply src/gameplay/voidGameTracker.ts]: Done in https://github.com/vck3000/ProAvalon/pull/623/commits/9a4498cbabcfc22221a241ce0d5d9239ed86be38

**Comment by kev306** [comment]
https://github.com/vck3000/ProAvalon/pull/623/commits/f918709dc1ed4d243de073ac4e6cf6719033e631 addresses issue 619

**Comment by vck3000** [comment]
This PR closes #619.

**Pull request merged**